revoke/revoke.go

```diff
@@ -123,69 +154,145 @@ func getIssuer(cert *x509.Certificate) *x509.Certificate {
 	}
 
 	return issuer
-
 }
 
-// check a cert against a specific CRL. Returns the same bool pair
-// as revCheck.
-func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
-	crl, ok := CRLSet[url]
+// checks whether CRL in memory is valid
+func (r *Revoke) isInMemoryCRLValid(key string) bool {
+	crl, ok := r.CRLSet[key]
 	if ok && crl == nil {
 		ok = false
-		delete(CRLSet, url)
+		r.CRLSetLck.Lock()
```

Comment:
RWMutex may improve efficiency, but placing the lock around map operation needs extra efforts. So far what I see is complicating this PR, which I thought the main focus is to run CRL check against a CRL file.